lib/AST/Decl.cpp

```diff
@@ -2092,9 +2092,42 @@ bool NominalTypeDecl::derivesProtocolConformance(ProtocolDecl *protocol) const {
     case KnownProtocolKind::BridgedNSError:
       return isObjC() && enumDecl->hasOnlyCasesWithoutAssociatedValues();
 
+    // Enums without associated values and enums with a raw type of String
+    // or Int can explicitly derive CodingKey conformance.
+    case KnownProtocolKind::CodingKey: {
+      Type rawType = enumDecl->getRawType();
+      if (rawType) {
+        auto parentDC = enumDecl->getDeclContext();
+        ASTContext &C = parentDC->getASTContext();
+
+        auto nominal = rawType->getAnyNominal();
+        return nominal == C.getStringDecl() || nominal == C.getIntDecl();
+      }
+
+      // Empty enums are allowed to conform as well.
+      return enumDecl->getAllElements().empty() ||
```

Comment:
Just a heads up: I'm changing `hasOnlyCasesWithoutAssociatedValues` so you no longer need the first part of this clause.